controllers/clusterenvironment_controller.go

```diff
@@ -177,6 +186,35 @@ func (r *ClusterEnvironmentReconciler) testConnection(ctx context.Context, cfg *
 	return nil
 }
 
+func (r *ClusterEnvironmentReconciler) reconcileServiceNamespaces(ctx context.Context, ce *primazaiov1alpha1.ClusterEnvironment) error {
+	errs := []error{}
+	serviceclassesList := primazaiov1alpha1.ServiceClassList{}
+	if err := r.List(ctx, &serviceclassesList, &client.ListOptions{Namespace: ce.Namespace}); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	// for _, servicenamespaces := range ce.Spec.ServiceNamespaces {
+	var serviceclassFilteredList []primazaiov1alpha1.ServiceClass
+	for _, serviceclass := range serviceclassesList.Items {
+		if envtag.Match(ce.Spec.EnvironmentName, serviceclass.Spec.Constraints.Environments) {
+			serviceclassFilteredList = append(serviceclassFilteredList, serviceclass)
+		}
+	}
+	for _, serviceclass := range serviceclassFilteredList {
+		if err := servicenamespace.PushToServiceNamespaces(ctx, serviceclass, *ce, r.Scheme, r.Mapper, r.Client); err != nil {
+			if strings.Contains(err.Error(), "already exists") {
+				continue
+			} else {
+				errs = append(errs,
+					fmt.Errorf("error pushing service class '%s' to cluster environment '%s': %w", serviceclass.Name, ce.Name, err))
+			}
```

Comment:
you can use `apierrors "k8s.io/apimachinery/pkg/api/errors".IsAlreadyExists`

```suggestion
			if !apierrors.IsAlreadyExists(err) {
				errs = append(errs,
					fmt.Errorf("error pushing service class '%s' to cluster environment '%s': %w", serviceclass.Name, ce.Name, err))
			}
```